File: Civitas_Ruby/lib/sorpresa_convertirEspeculador.rb
#!/bin/env ruby
# encoding: utf-8
require_relative "jugador_especulador.rb"
require_relative "sorpresa.rb"
require_relative "diario.rb"
require_relative "jugador.rb"
module Civitas
  class Sorpresa_ConvertirEspeculador < Sorpresa
    def initialize(fin)
      super()
      @fianza = fin
    end
    
    def aplicarAJugador(actual, todos)
      nuevo = Jugador_Especulador.nuevoEspeculador(todos[actual],@fianza)
      
           
      todos.delete_at(actual)
      todos.insert(actual, nuevo)
      
      informe(actual, todos)
    end
    
    
    def informe(actual, todos)
       indice = actual
      if(jugadorCorrecto(actual,todos))
        
        puts "Valor de ACTUAL ES: #{@nombre}"
        cadena = "#{@nombre} se ha convertido en especulador"
        Diario.instance.ocurre_evento(cadena)

      end
    end
    
  end
end

File: Civitas_GUI/src/civitas/Sorpresa_convertirEspeculador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author alejandro
 */
public class Sorpresa_convertirEspeculador extends Sorpresa{
    
    private int fianza;
    
    Sorpresa_convertirEspeculador(int fin){
        super();
        fianza = fin;
    }
    
    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        Jugador_Especulador nuevo = new Jugador_Especulador(todos.get(actual), fianza);
        
        for(int i = 0; i < nuevo.getPropiedades().size(); i++)
            nuevo.getPropiedades().get(i).actualizaPropietarioPorConversion(nuevo);
        
        todos.remove(actual);
        todos.add(actual, nuevo);
        
        this.informe(actual, todos);
    }
    
    @Override
    protected void informe(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual, todos))
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + " se ha convertido en un especulador");
    }
}
